modules/heat_conduction/src/userobjects/GapFluxModelPressureDependentConduction.C

```diff
@@ -0,0 +1,71 @@
+
+#include "GapFluxModelPressureDependentConduction.h"
+
+registerMooseObject("HeatConductionApp", GapFluxModelPressureDependentConduction);
+
+InputParameters
+GapFluxModelPressureDependentConduction::validParams()
+{
+  InputParameters params = GapFluxModelBase::validParams();
+
+  params.addClassDescription("Heat flux model across a closed gap to calculate the conductance "
+                             "between two solid materials");
+  params.addCoupledVar("temperature", "The name of the temperature variable");
+  params.addParam<Real>(
+      "scaling_coefficient",
+      1.0,
+      "The leading coefficient for the closed gap conductance value; used for tuning");
+  params.addRequiredParam<MaterialPropertyName>(
+      "primary_conductivity", "The thermal conductivity of the primary surface solid material");
+  params.addRequiredParam<MaterialPropertyName>(
+      "secondary_conductivity", "The thermal conductivity of the secondary surface solid material");
+  params.addRequiredCoupledVar("normal_pressure",
+                               "The name of the Lagrange multiplier that holds the normal contact "
+                               "pressure in mortar formulations");
+  params.addRequiredParam<MaterialPropertyName>(
+      "primary_hardness", "The hardness value of the primary surface material");
+  params.addRequiredParam<MaterialPropertyName>("secondary_hardness",
+                                                "The hardness of the secondary surface material");
+
+  return params;
+}
+
+GapFluxModelPressureDependentConduction::GapFluxModelPressureDependentConduction(
+    const InputParameters & parameters)
+  : GapFluxModelBase(parameters),
+    _primary_T(adCoupledNeighborValue("temperature")),
+    _secondary_T(adCoupledValue("temperature")),
+    _scaling(getParam<Real>("scaling_coefficient")),
+    _primary_conductivity(getNeighborADMaterialProperty<Real>("primary_conductivity")),
+    _secondary_conductivity(getADMaterialProperty<Real>("secondary_conductivity")),
+    _normal_pressure(coupledValueLower("normal_pressure")),
+    _primary_hardness(getNeighborADMaterialProperty<Real>("primary_hardness")),
+    _secondary_hardness(getADMaterialProperty<Real>("secondary_hardness"))
+{
+}
+
+ADReal
+GapFluxModelPressureDependentConduction::computeFlux() const
+{
+  // Check that the surfaces are in actual contact with the pressure:
+  if (_normal_pressure[0] <= 0.0)
+    return 0.0;
+
+  // calculate the harmonic means of the two material properties
+  const ADReal k_sum = _primary_conductivity[_qp] + _secondary_conductivity[_qp];
+  const ADReal k_harmonic = 2 * _primary_conductivity[_qp] * _secondary_conductivity[_qp] / k_sum;
+
+  const ADReal h_sum = _primary_hardness[_qp] + _secondary_hardness[_qp];
+  const ADReal h_harmonic = 2 * _primary_hardness[_qp] * _secondary_hardness[_qp] / h_sum;
+
+  return _scaling * k_harmonic * (_primary_T[_qp] - _secondary_T[_qp]) * _normal_pressure[0] /
```

Comment:
Why is here `_normal_pressure` evaluated at zero. As far as I can tell, the pressure should be interpolated as the conductivities and other parameters defined in the materials.

Reply:
so just use ` _normal_pressure[_qp] ` ? I think I must have convinced myself that the normal pressure was always the 0th entry, which of course only works when applying pressure in the x direction and is bad code in general

Reply:
Yeah, `_normal_pressure[_qp]`, it shouldn't be only the 0th entry. It's an interpolated quantity within the constraint (in a different situation we may be interested in the nodal value, such as in a nodal aux kernel). I think your results may vary slightly after this change.

Reply:
Thanks! and yeah, the results will probably vary in at least the large strain simulation, and probably away from the analytical solution. The small strain case is such a tiny different pressure that I don't expect to see much change....if I do, time for some more debugging